plugins/woocommerce/src/Admin/API/AI/Patterns.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Admin\API\AI;
+
+use Automattic\WooCommerce\Blocks\AI\Connection;
+use Automattic\WooCommerce\Blocks\AIContent\PatternsHelper;
+use Automattic\WooCommerce\Blocks\AIContent\UpdatePatterns;
+use WP_Error;
+use WP_HTTP_Response;
+use WP_REST_Request;
+use WP_REST_Response;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Patterns controller
+ *
+ * @internal
+ */
+class Patterns extends AIEndpoint {
+	/**
+	 * Endpoint.
+	 *
+	 * @var string
+	 */
+	protected $endpoint = 'patterns';
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		$this->register(
+			array(
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'update_patterns' ),
+					'permission_callback' => array( Middleware::class, 'is_authorized' ),
+					'args'                => array(
+						'business_description' => array(
+							'description' => __( 'The business description for a given store.', 'woocommerce' ),
+							'type'        => 'string',
+						),
+						'images'               => array(
+							'description' => __( 'The images for a given store.', 'woocommerce' ),
+							'type'        => 'object',
+						),
+					),
+				),
+				array(
+					'methods'             => \WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'delete_patterns' ),
+					'permission_callback' => array( Middleware::class, 'is_authorized' ),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Update patterns with the content and images powered by AI.
+	 *
+	 * @param  WP_REST_Request $request Request object.
+	 *
+	 * @return WP_Error|WP_HTTP_Response|WP_REST_Response
```

Comment:
Can we remove `WP_HTTP_Response`? Seems like this method and `delete_patterns` should have the same signatures and both return `WP_Error` or `WP_REST_Response`?

Reply:
Good catch! Fixed!